### ISSUE TITLE
Using snakemake -q all hides SLURM error messages

### ISSUE THREAD

**Issue by peterjc** [snakemake/snakemake#3173]
**Snakemake version**

```
$ snakemake --version
8.24.0
```

**Describe the bug**

I am using snakemake (actually via Python) on Linux but find if run in quite mode (``--quiet all``) this does not report SLURM error messages.

**Minimal example**

This example fails with a clear error message (there is no profile setup with SLURM settings):

```
$ rm -rf *.md5; \
  snakemake --executor slurm -s demo.smk -p $(for f in *.fna; do echo $f.md5; done); \
  echo Finished with return code 0
Error: maximum number of parallel jobs/used nodes has to be specified for remote execution (use --jobs N with N being a number >= 1)
Finished with return code 0
```

This is a clear actionable failure message.

(There is a separate problem here in that snakemake ought to have given a non-zero return code)

However, if I add ``--quiet all`` or ``-q all`` this hides the error message about needing to supply ``--jobs N``:

```
$ rm -rf *.md5; \
  snakemake --executor slurm -q all -s demo.smk -p $(for f in *.fna; do echo $f.md5; done); \
  echo Finished with return code 0
Finished with return code 0
```

For reference, adding ``--jobs 10`` then seems to start but waits indefinitely, without quiet mode at least there is a clue: ``No SLURM account given, trying to guess.``

**Additional context**

The example ``demo.smk`` is a per my recent report issue #3152 about quiet mode not being quiet enough.

**Comment by cmeesters** [comment]
hm, I cannot reproduce your zero error code:

```shell
$ snakemake .... # all settings in the profile, no special flags
...
Error: maximum number of parallel jobs/used nodes has to be specified for remote execution (use --jobs N with N being a number >= 1)
$ echo $?
1
```
If run with `-q`, it gives the same error and exit code. With `-q all` the error message is suppressed and the exit code remains. With `-q ""` (the empty string example), I get the same error and exit code as you.

The executor will attempt to submit without an account. A cluster with only default accounts is considered one with an unorthodox setup, hence the message. If your jobs are correctly submitted and kept in the queue / partition, the reason for it might be outside of Snakemake's influence. This is something we might have to discuss or investigate.

This leaves the command line interface help message. To me, `[...]` means: “May be omitted” / anything within the square brackets is optional. Like in `[--configfile FILE [FILE ...]]`, where one file is mandatory, the other(s) not. When you indicate an empty string, you do give an argument, albeit not a valid one, hence the error message. Does this make sense?

You write, that

> Specifically the choices does no make it clear that it can alternatively have no argument

Yet, the quoted help message states

> If used without arguments, do not output any progress or rule information.

So, no argument is fine. Then the run will be quite, except for warnings and errors.

**Comment by peterjc** [comment]
Some tools will accept an empty string, but that or no argument both seem confusing to me. Getting back to:

```console
% snakemake --quiet "" ; echo "Return code $?"
...
snakemake: error: argument --quiet/-q: invalid choice: '' (choose from 'all', 'host', 'progress', 'rules')
Return code 2
```

Can you tweak the error message for a bad option to say something like:

```
snakemake: error: argument --quiet/-q: invalid choice: '' (choose from 'all', 'host', 'progress', 'rules', or no argument at all)
```

Or (and I think this would be clearest), give the no-argument "do not output any progress or rule information" an alias like "silent"? Then you can list "silent" in the command line help explicitly (and retain the no argument behaviour for backward compatibility).

**Comment by peterjc** [comment]
Apologies. Embarrassingly it looks like at some point in my testing copy-and-pasting commands, ``$?`` got evaluated and replaced with ``0``, so the echo command was literally ``echo Finished with return code 0`` 😞

I'm writing this from a different machine, but that more than likely explains things. Opps. Here the tests are working, in that they return a non-zero value:

```console
% rm -rf *.md5; \
  snakemake --executor slurm -s demo.smk -p $(for f in *.fna; do echo $f.md5; done); \
  echo "Return code $?"
Error: maximum number of parallel jobs/used nodes has to be specified for remote execution (use --jobs N with N being a number >= 1)
Return code 1
```

```console
% rm -rf *.md5; \
  snakemake --executor slurm -q all -s demo.smk -p $(for f in *.fna; do echo $f.md5; done); \
  echo "Return code $?"
zsh: no matches found: *.md5
Return code 1
```

(zsh is more verbose about non-matching wildcard expansions than bash)

**Comment by cmeesters** [comment]
> Some tools will accept an empty string, but that or no argument both seem confusing to me.

I'm afraid, but this is due to the default behaviour from the `argparse` module in Python. If a tool accepts an empty string, that tool still receives an argument from the command line. But, yes, we could adapt the handling. I will take it on me to code a proposal. But understand, that it's not high on my list of priorities at the moment. If you want to go ahead, please let me know of your PR.

As to your last posts: That is actually matching my observation — regardless of verbosity, we get the same error code. And, of course. Is your concern, that the error message itself is suppressed, too?

**Comment by peterjc** [comment]
Yes, I don't think the error message should get hidden - in this example:

``
Error: maximum number of parallel jobs/used nodes has to be specified for remote execution (use --jobs N with N being a number >= 1)
``

**Comment by peterjc** [comment]
Just found this old issue after a search because I've got a snakemake run failing with `Error: maximum number of parallel jobs/used nodes has to be specified for remote execution (use --jobs N with N being a number >= 1)` - so the good news is the error wasn't silenced anymore, and I'm going to close this as apparently fixed.

(I now need to work out why this particular configuration of mine isn't setting the number of jobs...)

**Comment by cmeesters** [comment]
@peterjc oh, I am so sorry, I missed following up entirely.

**Comment by peterjc** [comment]
Not to worry - it hadn't been an problem for me, but I've been switching projects and using that code again this week.
